src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
> The only case I could find it different now is break with a label:

brake with label should be moved to separate issue to avoid overcomplication of this PR. 
Provided example is it compilable ? I found http://stackoverflow.com/a/14960484/1015848 , we should consider only compilable code.
Lets move discussion of this weird cases in separate issue.